lib/astutils.cpp

```diff
@@ -238,6 +238,16 @@ bool isSameExpression(bool cpp, bool macro, const Token *tok1, const Token *tok2
     return commutativeEquals;
 }
 
+bool equalTokValue(const Token * const tok1, const Token * const tok2)
+{
+    return tok1->values() == tok2->values() && tok1->values().size()==1U && tok1->values().front().isKnown();
+}
+
+bool notEqualTokValue(const Token * const tok1, const Token * const tok2)
```

Comment:
how about the name `isDifferentKnownValues` instead?

We have a `Token::hasKnownIntValue()` would that be enough for you or would you like to handle other types of values also? Then maybe a `Token::hasKnownValue()` could be added.

Then you could rewrite this function as:

    return tok1->hasKnownValue() && tok2->hasKnownValue() && tok1->values() != tok2->values();